validator/beaconnode/src/test/java/tech/pegasys/teku/validator/beaconnode/GenesisDataProviderTest.java

```diff
@@ -23,79 +23,107 @@
-    verify(validatorApiChannel).getGenesisTime();
+    when(validatorApiChannel.getGenesisData())
+        .thenReturn(
+            SafeFuture.completedFuture(
+                Optional.of(new GenesisData(GENESIS_TIME, GENESIS_VALIDATORS_ROOT))));
+    assertThat(genesisDataProvider.getGenesisData())
+        .isCompletedWithValue(new GenesisData(GENESIS_TIME, GENESIS_VALIDATORS_ROOT));
+    verify(validatorApiChannel).getGenesisData();
 
     // Subsequent requests just return the cached version
-    assertThat(genesisTimeProvider.getGenesisTime()).isCompletedWithValue(GENESIS_TIME);
+    assertThat(genesisDataProvider.getGenesisData())
+        .isCompletedWithValue(new GenesisData(GENESIS_TIME, GENESIS_VALIDATORS_ROOT));
     verifyNoMoreInteractions(validatorApiChannel);
   }
 
   @Test
   void shouldRetryWhenGenesisTimeFailsToLoad() {
-    when(validatorApiChannel.getGenesisTime())
+    when(validatorApiChannel.getGenesisData())
         .thenReturn(failedFuture(new RuntimeException("Nope")))
-        .thenReturn(completedFuture(Optional.of(GENESIS_TIME)));
+        .thenReturn(
+            completedFuture(Optional.of(new GenesisData(GENESIS_TIME, GENESIS_VALIDATORS_ROOT))));
 
     // First request fails
-    final SafeFuture<UInt64> result = genesisTimeProvider.getGenesisTime();
-    verify(validatorApiChannel).getGenesisTime();
+    final SafeFuture<UInt64> result = genesisDataProvider.getGenesisTime();
+    verify(validatorApiChannel).getGenesisData();
     assertThat(result).isNotDone();
     Assertions.assertThat(asyncRunner.hasDelayedActions()).isTrue();
 
     // Retry is scheduled.
     asyncRunner.executeQueuedActions();
-    verify(validatorApiChannel, times(2)).getGenesisTime();
+    verify(validatorApiChannel, times(2)).getGenesisData();
     assertThat(result).isCompletedWithValue(GENESIS_TIME);
   }
 
   @Test
   void shouldRetryWhenGenesisTimeIsNotYetKnown() {
-    when(validatorApiChannel.getGenesisTime())
+    when(validatorApiChannel.getGenesisData())
         .thenReturn(completedFuture(Optional.empty()))
-        .thenReturn(completedFuture(Optional.of(GENESIS_TIME)));
+        .thenReturn(
+            completedFuture(Optional.of(new GenesisData(GENESIS_TIME, GENESIS_VALIDATORS_ROOT))));
 
     // First request fails
-    final SafeFuture<UInt64> result = genesisTimeProvider.getGenesisTime();
-    verify(validatorApiChannel).getGenesisTime();
+    final SafeFuture<UInt64> result = genesisDataProvider.getGenesisTime();
+    verify(validatorApiChannel).getGenesisData();
     assertThat(result).isNotDone();
     Assertions.assertThat(asyncRunner.hasDelayedActions()).isTrue();
 
     // Retry is scheduled.
     asyncRunner.executeQueuedActions();
-    verify(validatorApiChannel, times(2)).getGenesisTime();
+    verify(validatorApiChannel, times(2)).getGenesisData();
     assertThat(result).isCompletedWithValue(GENESIS_TIME);
   }
+
+  @Test
+  void shouldRetryWhenGenesisValidatorsRootIsNotYetKnown() {
```

Comment:
This case is the same as `shouldRetryWhenGenesisTimeIsNotYetKnown`

Reply:
yep that's what i copied, i wanted to show that validatorsRoot would work, i'll think of something trivial

Reply:
But both are just testing `getGenesisData`.  I don't think we need to individually check that the shortcut `getGenesisTime` and `getGenesisValidatorRoot` methods work - they are a trivial map from `getGenesisData`.